components/ILIAS/Mail/classes/Attachments/FileDataRCHandling.php

```diff
@@ -0,0 +1,141 @@
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+declare(strict_types=1);
+
+use ILIAS\Filesystem\Stream\Streams;
+use ILIAS\ResourceStorage\Identification\ResourceCollectionIdentification;
+use ILIAS\ResourceStorage\Collection\ResourceCollection;
+
+trait FileDataRCHandling
+{
+    /**
+     * @param array                       $path_to_files
+     * @param ilMailAttachmentStakeholder $stakeholder
+     * @return ResourceCollection
+     */
+    protected function getCurrentCollection(
+        array $path_to_files,
+        ilMailAttachmentStakeholder $stakeholder
+    ): \ILIAS\ResourceStorage\Collection\ResourceCollection {
+        set_error_handler(static function ($severity, $message, $file, $line): void {
+            throw new ErrorException($message, $severity, 0, $file, $line);
+        });
+
+        try {
+            global $DIC;
+            $file_system = $DIC->filesystem()->storage();
+            $rcid = $this->storage->collection()->id();
+            $collection = $this->storage->collection()->get($rcid);
+            foreach ($path_to_files as $path_to_file) {
+                $path_to_file = str_replace('/var/ilias/data/ilias', '', $path_to_file);
+                $rid = $this->storage->manage()->stream(
+                    $file_system->readStream($path_to_file),
+                    $stakeholder,
+                    md5(basename($path_to_file))
+                );
+                $collection->add($rid);
+            }
+            $this->storage->collection()->store($collection);
+        } catch (Exception $e) {
+            throw new Exception("Storing file into collection failed: " . $e->getMessage());
+        } finally {
+            restore_error_handler();
+        }
+
+        return $collection;
+    }
+
+    /**
+     * @param array $mail_data
+     * @return array
+     */
+    public function FilesFromLegacyToIRSS(array $mail_data): array
+    {
+        $files = [];
+        $path_to_files = [];
+        $attachment = unserialize($mail_data['attachments']->__toString(), ['allowed_classes' => false]);
+        foreach ($attachment as $file) {
+            $path_to_files[] = $this->fdm->getAbsoluteAttachmentPoolPathByFilename($file);
+        }
+        $collection = $this->getCurrentCollection($path_to_files, new ilMailAttachmentStakeholder());
+        foreach ($collection->getResourceIdentifications() as $rcid) {
+            $files[] = $rcid->serialize();
+        }
+
+        return $files;
+    }
+
+    /**
+     * @param array $mail_data
+     * @return ResourceCollectionIdentification|null
+     */
+    public function getIDforCollection(array $mail_data): ?ResourceCollectionIdentification
```

Comment:
We should rename the method to `getIDforCollection` (with a lowercase `d`):

```suggestion
    public function getIdforCollection(array $mail_data): ?ResourceCollectionIdentification
```